cmd/jaeger/main.go

```diff
@@ -28,3 +32,14 @@ func main() {
 		log.Fatal(err)
 	}
 }
+
+func featuresCommand() *cobra.Command {
+	return &cobra.Command{
+		Use:   "features",
+		Short: "Displays the list of supported features",
+		Long:  "The 'features' command shows all supported features in the current build of Jaeger.",
+		Run: func(_ *cobra.Command, _ []string) {
+			features.DisplayFeatures()
```

Comment:
consider this instead
```suggestion
			fmt.Println(features.DisplayFeatures())
```

Reply:
Instead of that, I am thinking of returning an array of strings from DisplayFeatures(). 

In DisplayFeatures(), we would be printing everything as it is. The use of the array returned is only for testing. 

In the array, we would be having the id and we would be checking if the id is not null.

Does it sound okay?

Reply:
what's the point of the array? The command needs to print everything at once.

Reply:
The point of the array is for just testing. In the display features, I would still be printing as it is doing now. 

The only difference would be that I would be the following:

```
func DisplayFeatures() []string{
f := featuregate.GlobalRegistry()
	if f == nil {
		return
	}
        var ids []string
	f.VisitAll(func(g *featuregate.Gate) {
                 ids = append(ids, g.ID()) //append the new ids
		fmt.Printf("Feature:\t%s\n", g.ID())
		if !g.IsEnabled() {
			fmt.Printf("Default state:\t%s\n", "On")
		} else {
			fmt.Printf("Default state:\t%s\n", "Off")
		}
		fmt.Printf("Description:\t%s\n", g.Description())

		if ref := g.ReferenceURL(); ref != "" {
			fmt.Printf("ReferenceURL:\t%s\n", ref)
		}
		fmt.Println()
	})
        return ids // return the ids here
}
```
and in the testing, we can check if there is some id that is null or not. We can do the same thing with description because these two are the only attributes of a feature flag that are not empty.

Reply:
overcomplicating

Reply:
> overcomplicating

Other solution is to simply return the final string. But what all parameters shall I check in that? 

One thing I can think of is just the ID of the flags. If id of any of them is empty, we can throw an error.
Another thing we can check is the description. If description of any of them is empty, we can throw an error.

Is this method fine?

Reply:
https://github.com/jaegertracing/jaeger/pull/6542#issuecomment-2605593963